glue_jupyter/bqplot/profile/tests/test_viewer.py

```diff
@@ -157,3 +161,80 @@ def test_unit_conversion(app):
 
     assert len(d2.subsets) == 1
     assert_equal(d2.subsets[0].to_mask(), [0, 1, 1])
+
+
+def test_composite_xrange(app):
+    session = app.session
+
+    wcs2 = WCS(naxis=1)
+    wcs2.wcs.ctype = ['WAVE']
+    wcs2.wcs.crval = [10]
+    wcs2.wcs.cdelt = [10]
+    wcs2.wcs.crpix = [1]
+    wcs2.wcs.cunit = ['cm']
+
+    d2 = Data(f2=[2000, 1000, 3000])
+    d2.get_component('f2').units = 'mJy'
+    d2.coords = wcs2
+
+    data_collection = session.data_collection
+    data_collection.append(d2)
+
+    viewer = app.profile1d(data=d2)
+    tool = viewer.toolbar.tools['bqplot:xrange']
+    tool.activate()
+    tool.interact.brushing = True
+    tool.interact.selected = [15, 35]
+    tool.interact.brushing = False
+
+    session.edit_subset_mode.mode = AndNotMode
+
+    tool.interact.brushing = True
+    tool.interact.selected = [25, 45]
+    tool.interact.brushing = False
+    tool.deactivate()
+
+    sbst = session.data_collection.subset_groups[0].subset_state
+    assert isinstance(sbst, AndState)
+    assert isinstance(sbst.state1, RangeSubsetState)
```

Comment:
Without this patch, the two new tests would fail like this (I ran it on current `main`):

```
>       assert isinstance(sbst.state1, RangeSubsetState)
E       assert False
E        +  where False = isinstance(<glue.core.subset.AndState ...>, RangeSubsetState)
E        +    where <glue.core.subset.AndState ...> = <glue.core.subset.AndState ...>.state1
```

But the tests should pass here.